instructions.ts

```diff
@@ -8,14 +8,19 @@ import type {
   Task,
 } from "./types.ts";
 
-import { evaluate, reset, shift } from "./deps.ts";
+import { reset, shift } from "./deps.ts";
 import { createFrameTask } from "./run/frame.ts";
+import { createObservable } from "./run/observer.ts";
 
 export function suspend(): Operation<void> {
   return {
     *[Symbol.iterator]() {
-      return yield function Suspend() {
-        return shift<Result<void>>(function* () {});
+      return yield function Suspend(_, signal) {
+        return shift<Result<void>>(function* (k) {
+          if (signal.aborted) {
+            k({ type: "resolved", value: void 0 });
+          }
```

Comment:
Now, if a block of instructions has been aborted, a suspend will just resume immediately.

Reply:
k

(I promise that will be the only one)